sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client_parser.c

```diff
@@ -130,13 +156,14 @@ static void test_az_iot_provisioning_client_received_topic_payload_parse_assigne
 }
 
 static void
-test_az_iot_provisioning_client_received_topic_payload_parse_invalid_certificate_error_succeed()
+test_az_iot_provisioning_client_parse_received_topic_and_payload_invalid_certificate_error_succeed()
 {
   az_iot_provisioning_client client;
   az_span received_topic = AZ_SPAN_FROM_STR("$dps/registrations/res/401/?$rid=1");
-  az_span received_payload = AZ_SPAN_FROM_STR(
-      "{\"errorCode\":401002,\"trackingId\":\"" TEST_ERROR_TRACKING_ID "\","
-      "\"message\":\"" TEST_ERROR_MESSAGE_INVALID_CERT "\",\"timestampUtc\":\"" TEST_ERROR_TIMESTAMP "\"}");
+  az_span received_payload
+      = AZ_SPAN_FROM_STR("{\"errorCode\":401002,\"trackingId\":\"" TEST_ERROR_TRACKING_ID "\","
```

Comment:
Add test with extra properties in the JSON to validate we are skipping them (if one already doesn't exist).

Reply:
Also, do we care about the behavior of having repeated properties in the JSON? It looks like for us, first one wins and we ignore subsequent ones.

For example: "{errorCode: 1, errorCode: 2, ... }"

Reply:
Won't add to this PR. The goal here was to get 100% coverage based on current AzSDK tools.
See my other PR enabling branch coverage: based on that we should definitely add the cases you mention and others (future PR though).

Reply:
Also, I'd like to dig into the protocol itself more and uncover actual possible JSON messages returned from the service today (higher pri than the other tests).